biz.aQute.repository/src/aQute/bnd/repository/maven/pom/provider/BndPomRepository.java

```diff
@@ -88,13 +99,57 @@ public synchronized void init() {
 				repository.close();
 				throw new IllegalStateException("We have neither a pom, revision, or query set!");
 			}
+
 			bridge = new BridgeRepository(repoImpl);
 			inited = true;
 		} catch (Exception e) {
 			throw Exceptions.duck(e);
 		}
 	}
 
+	private void startPoll() {
+		Workspace ws = registry.getPlugin(Workspace.class);
+		if ((ws != null) && (ws.getGestalt().containsKey(Constants.GESTALT_BATCH)
+				|| ws.getGestalt().containsKey(Constants.GESTALT_CI)
+				|| ws.getGestalt().containsKey(Constants.GESTALT_OFFLINE))) {
+			return;
+		}
+		final AtomicBoolean busy = new AtomicBoolean();
+		pomPoller = Processor.getScheduledExecutor().scheduleAtFixedRate(new Runnable() {
+
+			@Override
+			public void run() {
+				if (busy.getAndSet(true))
+					return;
+				try {
+					boolean needsUpdate = false;
+					for (Entry<URI,Long> pomFilesEntry : pomFiles.entrySet()) {
+						if (pomFilesEntry.getValue() == null) {
+							// not interested
+							continue;
+						}
+						File f = new File(pomFilesEntry.getKey());
+						if (f != null && f.isFile() && f.lastModified() > pomFilesEntry.getValue()) {
+							needsUpdate = true;
+							pomFilesEntry.setValue(f.lastModified());
+						}
+					}
+					if (needsUpdate) {
+						refresh();
+						for (RepositoryListenerPlugin listener : registry.getPlugins(RepositoryListenerPlugin.class))
+							listener.repositoryRefreshed(BndPomRepository.this);
+					}
+
+				} catch (Exception e) {
+					reporter.error("Error when polling for %s for change", this);
+				} finally {
+					busy.set(false);
+				}
+			}
+
+		}, 5000, 5000, TimeUnit.MILLISECONDS);
```

Comment:
Rather than having a fixed poll time and no way to disable polling, the polling time should be configurable. https://github.com/bndtools/bnd/blob/master/biz.aQute.repository/src/aQute/bnd/repository/osgi/OSGiRepository.java#L134-L152